### PR TITLE
Run fish_indent on all fish scripts

### PR DESCRIPTION
**Pull request by faho**
If we want to enforce indentation, we should do it like our tool does it.

### CODE REVIEW

**Comment by ridiculousfish** [comment]
`git diff --word-diff=color -b <branch>` is how I'm reviewing this

**Comment by faho** [comment]
This is pretty much an extensive test for fish_indent, and I have to say it does some things I'm not too sure of:

- Remove spaces after pipes (e.g. `cmd help|sed -ne` in make_vcs_completions_generic.fish)

- Un-linebreak "while read", creating a 173 char long line in completions/busctl.fish (it breaks other lines after 72 chars)

And the thing I don't really care about, but others might:

- Remove spaces after redirection characters (e.g. "^/dev/null" or ">/dev/null")

File: share/completions/emerge.fish [hunk 1]
```diff
@@ -3,29 +3,26 @@
...
- end
+    if test -d /var/db/pkg
+        find /var/db/pkg/ -type d | cut -d'/' -f5-6 | sort | uniq | sed 's/-[0-9]\{1,\}\..*$//' | sed -e '/^ *$/d'
+        return
+    end
 end
 
 function __fish_emerge_print_all_pkgs --description 'Prints completions for all available packages on the system from /usr/portage'
- if test -d /usr/portage
-   find /usr/portage/ -maxdepth 2 -type d | cut -d'/' -f4-5 | \
```
Comment (ridiculousfish) [inline share/completions/emerge.fish]: Unfortunate to lose this nie wrapping

File: share/completions/wesnoth.fish [hunk 2]
```diff
@@ -28,7 +28,7 @@ complete -c wesnoth -l exit-at-end --description 'Exits once the scenario is ove
 #complete -c wesnoth -l controller<number> --description 'Selects the controller for this side'
 complete -c wesnoth -l era --description 'Use this option to play in the selected era instead of the "Default" era'
 complete -c wesnoth -l nogui --description 'Runs the game without the GUI'
-complete -c wesnoth -l parm<number> --description 'Sets additional parameters for this side'
```
Comment (ridiculousfish) [inline share/completions/wesnoth.fish]: Oh dear....

File: share/completions/iptables.fish [hunk 3]
```diff
@@ -1,103 +1,104 @@
...
+    # This only works as root, so ignore errors
+    iptables $tablearg -L ^ /dev/null | grep Chain | while read a b c
+        echo $b
+    end
 end
 
 
 function __fish_iptables_chains
-	set -l table (__fish_iptables_current_table)
-	[ -z $table ]; and set -l table "*"
```
Comment (ridiculousfish) [inline share/completions/iptables.fish]: Unfortunate to break this 'and' chaining across two lines

File: share/completions/wesnoth.fish (hunk 2 above)
Comment (faho) [inline share/completions/wesnoth.fish]: I'm pretty sure we should rename that parameter - I'm guessing this was supposed to be a poor man's hint. Still unfortunate fish_indent breaks (assuming the completion works at all).

File: share/completions/iptables.fish (hunk 3 above)
Comment (faho) [inline share/completions/iptables.fish]: Yeah - how about only breaking to achieve line length? With the possible exception of "if; begin; a; and; b; end;" blocks.

File: share/completions/wesnoth.fish (hunk 2 above)
Comment (ridiculousfish) [inline share/completions/wesnoth.fish]: The completion is definitely broken.

**Comment by ridiculousfish** [comment]
I think I want fish_indent to not put a space after redirections (except pipes):

    eval command $cmd --version > /dev/null ^ /dev/null
    if not tty 0>& 1 > /dev/null

would look better as

    eval command $cmd --version >/dev/null ^/dev/null
    if not tty 0>&1 >/dev/null

**Comment by faho** [comment]
>I also think it should prefer to put and conditions on the same line:

I prefer the third, but hate the second (and would take the first over the third).

My reason is that I dislike long lines - heck, I need to scroll the second one right now, in this browser window (it only goes to "$debi").

What I like about the third is that the condition-end is put on a different level than the body of the if-branch, making it obvious that it's belonging to the condition.

What however absolutely should not be done is breaking up `[ -z $var ]; and set var $value` - a rule of "one combiner per line" should presumably work in all cases.

>I think I want fish_indent to not put a space after redirections (except pipes):

I don't care too much about that, and I'd be okay if fish_indent (which does a lot more than just indent) would leave it alone.

File: share/completions/wesnoth.fish (hunk 2 above)
Comment (faho) [inline share/completions/wesnoth.fish]: This was fixed in 6157a9a85890d79521e77cb4e51d4d52793ac516.

**Comment by faho** [comment]
I'm closing this since it's not intended to be merged, but it's a nice reminder what should be fixed in fish_indent.

**Comment by zanchey** [comment]
I was reminded of this when reading http://journal.stuffwithstuff.com/2015/09/08/the-hardest-program-ive-ever-written/

**Pull request closed without merging**